provider/src/iosMain/kotlin/at/asitplus/crypto/provider/os/IOSKeychainProvider.kt

```diff
@@ -186,9 +193,13 @@ sealed class iosSigner<H : unlockedIOSSigner>(
                         is ECDSA -> kSecAttrKeyTypeEC
                         is RSA -> kSecAttrKeyTypeRSA
                     },
+                    kSecMatchLimit to kSecMatchLimitOne,
                     kSecReturnRef to true,
 
-                    kSecUseOperationPrompt to config.unlockPrompt.v.message,
+                    kSecUseAuthenticationContext to LAContext().apply {
```

Comment:
we will need to go down two routes here: on the on hand keeping the context and emulating Android-Style timeout as in my original swift-based approach, so that key with biometric unlock works the same on both platform with and without timeout. on on the other hand, allow for low-level customisation as in my original swift approach, so all bases are covered.

as for attestation: In the generic case, fix it to P256 and P384 so both platforms behave the same and on android,  i think we can already freely customise, because this is anyway a superset of what we can emulate on iOS

Reply:
Also: While the Secure enclave does not support RSA, the keychain can protect access to those keys using biometrics and passcode to behave the same as enclave-protected key material, so generating RSA keys and protecting them using passcode/biometrics should work. too.

Reply:
This is already possible by explicitly opting out of SE protection (`tpm { requireSecureEnclave = false }`).

I am reluctant to transparently "downgrade" the expected security without explicit opt-in.

Reply:
> keeping the context and emulating Android-Style timeout as in my original swift-based approach, so that key with biometric unlock works the same on both platform with and without timeout

this is something i am very conflicted on... on the one hand, it's definitely _needed_; on the other hand, we are pretending there's security where there's none... a malicious app could reuse our key an arbitrary number of times with a single biometry unlock, even if we set a small/zero reuse timeout in our own code. this is purely theater, the security isn't being enforced anywhere.

Reply:
> This is already possible by explicitly opting out of SE protection (`tpm { requireSecureEnclave = false }`).
> 
> I am reluctant to transparently "downgrade" the expected security without explicit opt-in.

D'Accord!

Reply:
> > keeping the context and emulating Android-Style timeout as in my original swift-based approach, so that key with biometric unlock works the same on both platform with and without timeout
> 
> this is something i am very conflicted on... on the one hand, it's definitely _needed_; on the other hand, we are pretending there's security where there's none... a malicious app could reuse our key an arbitrary number of times with a single biometry unlock, even if we set a small/zero reuse timeout in our own code. this is purely theater, the security isn't being enforced anywhere.

I partially agree. The weird thing is, you can actually get that timeout behaviour enforced by the SE, but only if you recently **unlocked** your device. Completely bonkers on Apple's side. Moreover? What is your scenario for a malicious app? It will mos def not come from the app store and if you jailbreak your device all bets are off anyways. 

In addition, you can and want to add another layer of theater through attestation and this is as good as it gets on iOS.